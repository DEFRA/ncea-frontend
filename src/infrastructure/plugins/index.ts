export { customHapiPino } from './logger';
export { customHapiRoutes } from './router';
export { customHapiViews } from './views';

