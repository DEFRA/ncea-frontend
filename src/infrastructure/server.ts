'use strict';

import { Config } from '../config/environmentConfig';
import Hapi, { Server } from '@hapi/hapi';

// Create the hapi server
const server: Server = Hapi.server({
  host: process.env.HOST ?? 'localhost',
  port: Config.env !== 'test' ? Config.port : 4000,
  routes: {
    validate: {
      options: {
        abortEarly: false,
      },
    },
  },
});

const initializeServer = async (): Promise<Server> => {
  // Register vendors plugins
  await server.register([require('@hapi/inert'), require('@hapi/vision')]);

  // Register the custom plugins
  await server.register([
    require('./plugins/shared_data'),
    require('./plugins/views'),
    require('./plugins/router'),
    require('./plugins/logger'),
  ]);

  await server.initialize();
  return server;
};

const startServer = async (): Promise<Server> => {
  await server.start();
  console.log(`Server running at: ${server.info.uri}`);
  return server;
};

export { initializeServer, startServer };
