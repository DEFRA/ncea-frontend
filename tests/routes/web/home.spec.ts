'use strict';

import { Server } from '@hapi/hapi';

import { createServer } from '../../../src/infrastructure/server';

describe('Home Routes', () => {
  let server: Server;

  beforeAll((done) => {
    createServer().then((s: Server) => {
      server = s;
      done();
    });
  });

  afterAll((done) => {
    server.stop().then(() => done());
  });

  it('should GET / route works', async () => {
    const options = {
      method: 'GET',
      url: '/',
    };

    const response = await server.inject(options);
    expect(response.statusCode).toEqual(200);
    expect(response.result).toEqual('Hello, Natural Capital Search Service');
  });
});
