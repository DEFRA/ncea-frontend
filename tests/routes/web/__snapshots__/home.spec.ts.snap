// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Home > Sanpshot verification should match the home screen snapshot 1`] = `
"<!DOCTYPE html>
<html lang="en" class="govuk-template app-html-class">
  <head>
    <meta charset="utf-8">
    <title>
  Natural Capital Search Service - GOV.UK
</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, viewport-fit=cover">
    <meta name="theme-color" content="#0b0c0c">

    
  
      <link rel="icon" sizes="48x48" href="/assets/images/favicon.ico">
      <link rel="icon" sizes="any" href="/assets/images/favicon.svg" type="image/svg+xml">
      <link rel="mask-icon" href="/assets/images/govuk-icon-mask.svg" color="#0b0c0c">
      <link rel="apple-touch-icon" href="/assets/images/govuk-icon-180.png">
      <link rel="manifest" href="/assets/manifest.json">
    


    
  <link rel="stylesheet" href="/assets/application.css">

    
  </head>
  <body class="govuk-template__body app-body-class">
    <script>document.body.className += ' js-enabled' + ('noModule' in HTMLScriptElement.prototype ? ' govuk-frontend-supported' : '');</script>
    

    
      <a href="#main-content" class="govuk-skip-link" data-module="govuk-skip-link">Skip to main content</a>

    

    
  <header class="header-container">
    <div class="header-div-container header-main">
      <div class="site-width-container">
        <div class="header-logo-container"><img alt="Department for Environment Food &amp; Rural Affairs Logo" src="/assets/images/defra_logo.svg" width="52" height="42" decoding="async" data-nimg="1" class="defra-logo" loading="lazy" style="color:transparent">
          <a target="_blank" class="header-logo-link logo-link" href="https://www.gov.uk/government/organisations/department-for-environment-food-rural-affairs">
            <span class="defra-logo-text">Department for Environment <br> Food &amp; Rural Affairs</span></a>
          <span class="govuk-header__service-name">Natural Capital Search Service</span></div>
      </div>
    </div>
  </header>


    
  
      <div class="govuk-width-container">
        
  <div class="govuk-phase-banner">
  <p class="govuk-phase-banner__content">
    <strong class="govuk-tag govuk-phase-banner__content__tag">
      Beta
    </strong>
    <span class="govuk-phase-banner__text">
      This is a new service – your <a class="govuk-link" href="#">feedback</a> will help us to improve it.
    </span>
  </p>
</div>


        <main class="govuk-main-wrapper app-main-class" id="main-content" role="main">
          
  <header class="full-width-banner" role="banner">
    <div class="banner-container">
      <h1 class="govuk-heading-xl banner-container__heading-xl">Natural Capital Search Service</h1>
      <span class="govuk-caption-l banner-container__caption-l">Use this service to search for data and information about the environment and ecosystems in the UK. This service can be used to support natural capital approaches to evidence and decision making</span>
    </div>
  </header>

  

<div class="govuk-grid-row">
  <div class="govuk-grid-column-full quick_search-container">
    <h1 class="govuk-heading-m quick_search-container__heading-m">
      
        Quick Search
      
    </h1>
    
      <span class="govuk-caption-m quick_search-container__caption-m">You can search by date, report title, theme or project codes, for example "seagrass beds, Porth Dinllaen, 2012".</span>
    
    <div class="search-block" data-module="search">
      <form action="/search" method="get" role="search" class="search-block__form govuk-!-margin-bottom-6 govuk-!-margin-top-4">

        

<div class="govuk-form-group">
  <label class="govuk-label search-block__label" aria-hidden="true" for="search">
    Search
  </label>

<div class="govuk-input__wrapper"><input class="govuk-input govuk-input govuk-!-margin-bottom-0 search-block__input" id="search" name="q" type="text" spellcheck="false" autocomplete="off" aria-controls="search-results" placeholder="Search"><div class="govuk-input__suffix" aria-hidden="true"><button type="submit" class="search-block__button">Search</button></div>
  </div>

</div>

      </form>
    </div>
  </div>
</div>

  <article class="govuk-grid-row" role="article">
    <div class="govuk-grid-column-full video-container">
      <h1 class="govuk-heading-m govuk-\\!-static-margin-0 video-container__heading-m">What is natural capital?</h1>
      <div class="video-container__frame">
        <iframe width="630" height="354" src="https://www.youtube.com/embed/4wx0rMruSJo?si=h-SD8VoSUSNwqo1S" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
      </div>
    </div>
  </article>

        </main>
      </div>
    


    
  <footer class="govuk-footer" role="contentinfo">
  <div class="govuk-width-container">
    
    <div class="govuk-footer__meta">
      <div class="govuk-footer__meta-item govuk-footer__meta-item--grow">
        
          <h2 class="govuk-visually-hidden">Support links</h2>
          
            <ul class="govuk-footer__inline-list">
              
                <li class="govuk-footer__inline-list-item">
                  <a class="govuk-footer__link" href="#">
                    Help
                  </a>
                </li>
              
                <li class="govuk-footer__inline-list-item">
                  <a class="govuk-footer__link" href="#">
                    Privacy
                  </a>
                </li>
              
                <li class="govuk-footer__inline-list-item">
                  <a class="govuk-footer__link" href="#">
                    Cookies
                  </a>
                </li>
              
                <li class="govuk-footer__inline-list-item">
                  <a class="govuk-footer__link" href="#">
                    Accessibility statement
                  </a>
                </li>
              
            </ul>
          
          
        
        <svg
          aria-hidden="true"
          focusable="false"
          class="govuk-footer__licence-logo"
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 483.2 195.7"
          height="17"
          width="41"
        >
          <path
            fill="currentColor"
            d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
          />
        </svg>
        <span class="govuk-footer__licence-description">
          
            All content is available under the
            <a
              class="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >Open Government Licence v3.0</a>, except where otherwise stated
          
        </span>
      </div>
      <div class="govuk-footer__meta-item">
        <a
          class="govuk-footer__link govuk-footer__copyright-logo"
          href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
        >© Crown copyright</a>
      </div>
    </div>
  </div>
</footer>



    
  
  <script type="module" src="/assets/all.js"></script>
  <script type="module">
    import {initAll} from '/assets/all.js'
    initAll()
  </script>

  </body>
</html>
"
`;
